server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -222,13 +259,19 @@ TaskCancellation getTaskCancellation(CancellableTask task) {
         for (TaskResourceUsageTracker tracker : taskResourceUsageTrackers) {
             Optional<TaskCancellation.Reason> reason = tracker.checkAndMaybeGetCancellationReason(task);
             if (reason.isPresent()) {
+                if (task instanceof SearchTask) {
+                    callbacks.add(tracker::incrementSearchTaskCancellations);
```

Comment:
Could be part of `SearchBackpressureTask` interface so each implementor (task) updates the right counter

Reply:
`incrementSearchTaskCancellations()` method cannot be moved to `SearchBackpressureTask` since it is coming from `TaskResourceUsageTracker` which is not specific to Task type.
However, updated the code the avoid use of `instanceof` in the code:
```
if (reason.isPresent()) {
    callbacks.add(() -> tracker.incrementTaskCancellations(task.getClass()));
    reasons.add(reason.get());
}
```

Reply:
It should not be moved, the `SearchBackpressureTask` could have method/hook which accepts `TaskResourceUsageTracker` as an argument and performs the update of the relevant counters.

Reply:
Makes sense. Updated the code.